src/particles/elements/Quad.H

```diff
@@ -58,17 +58,29 @@ namespace impactx
             amrex::ParticleReal const pt_ref = refpart.pt;
             amrex::ParticleReal const betgam2 = pow(pt_ref, 2) - 1.0_prt;
 
-            // advance position and momentum (drift)
-            p.pos(0) = cos(m_k*m_ds)*x + sin(m_k*m_ds)/m_k*px;
-            // px = px;
-            px = -m_k*sin(m_k*m_ds)*x + cos(m_k*m_ds)*px;
+            amrex::ParticleReal const omega = sqrt(abs(m_k));
 
-            p.pos(1) = cosh(m_k*m_ds)*y + sinh(m_k*m_ds)/m_k*py;
-            py = m_k*sinh(m_k*m_ds)*y + cosh(m_k*m_ds)*py;
+            if(m_k > 0.0) {
```

Comment:
A quick not on the `if`: There is no branch divergence here between different threads, but we can be more efficient in terms or GPU registers by avoiding the `if`.

What you can do to write a single code path is create a variable that stores the sign:
```c++
amrex::ParticleReal const sign = (m_k > 0.0) : -1.0_prt : 1.0_prt;
```

and multiply this always on the terms for `px` and `py` :)

The short-hand `if` I used above is called [ternary operator](https://www.programiz.com/cpp-programming/ternary-operator).

Reply:
I'm not sure how to avoid the "if" completely, because it is not only the sign but also cos -> cosh and sin -> sinh that changes in the expressions for px and py.

Reply:
Oh, I have overlooked that.

Reply:
In the future, I will split this class into k>0 & k<0 sub-classes.
This will allow us to avoid an `if` in kernel code for a condition we know upfront (and thus saves registers).

Not urgent now :)